### ISSUE TITLE
Find a way to integrate fiber dumps with test frameworks

### ISSUE THREAD

**Issue by djspiewak** [typelevel/cats-effect#3025]
Probably related to #2580. It's usually much more useful to have fiber dumps during unit testing, but the current solution is baked into `IOApp`. We need to find a composable way of integrating this into the test runtimes such as munit or cats-effect-testing.

**Comment by armanbilge** [comment]
Yes. I constantly need this and end up re-writing tests as scala-cli apps.

**Comment by djspiewak** [comment]
Exactly. I still am not totally sure what form this can take but I want it quite badly
